solr/core/src/java/org/apache/solr/cloud/api/collections/assign/DeleteReplicaRequest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cloud.api.collections.assign;
+
+/**
+ *
+ */
+public class DeleteReplicaRequest extends BaseAssignRequest {
```

Comment:
Likely a delete request should only specify the shard and let the plugin pick which replica to kill...
Otherwise no need to ask the plugin, just delete the replica directly.

Reply:
That's not always the case - when you want to decommission a node you know exactly what replicas to delete.

Reply:
Again, if it's just a delete and we already know what to delete, the plugin doesn't add much value.
Solr calls plugins to ask a question and the plugins provide answers. What's the question and what are the possible answers?